components/dashboard/src/contexts/FeatureFlagContext.tsx

```diff
@@ -1,145 +0,0 @@
-    const [enablePersonalAccessTokens, setPersonalAccessTokensEnabled] = useState<boolean>(false);
-    const [usePublicApiWorkspacesService, setUsePublicApiWorkspacesService] = useState<boolean>(false);
-    const [oidcServiceEnabled, setOidcServiceEnabled] = useState<boolean>(false);
-    const [orgGitAuthProviders, setOrgGitAuthProviders] = useState<boolean>(false);
-    const [userGitAuthProviders, setUserGitAuthProviders] = useState<boolean>(false);
-    const [newSignupFlow, setNewSignupFlow] = useState<boolean>(false);
-    const [linkedinConnectionForOnboarding, setLinkedinConnectionForOnboarding] = useState<boolean>(false);
-    const [paymentVerificationFlow, setPaymentVerificationFlow] = useState<boolean>(false);
-    const [orgOnlyAttribution, setOrgOnlyAttribution] = useState<boolean>(false);
-
-    useEffect(() => {
-        if (!user) return;
-        (async () => {
-            const featureFlags: FeatureFlagConfig = {
-                start_with_options: { defaultValue: false, setter: setStartWithOptions },
-                showUseLastSuccessfulPrebuild: { defaultValue: false, setter: setShowUseLastSuccessfulPrebuild },
-                personalAccessTokensEnabled: { defaultValue: false, setter: setPersonalAccessTokensEnabled },
-                publicApiExperimentalWorkspaceService: {
-                    defaultValue: false,
-                    setter: setUsePublicApiWorkspacesService,
-                },
-                oidcServiceEnabled: { defaultValue: false, setter: setOidcServiceEnabled },
-                // Default to true to enable on gitpod dedicated until ff support is added for dedicated
-                orgGitAuthProviders: { defaultValue: true, setter: setOrgGitAuthProviders },
-                userGitAuthProviders: { defaultValue: false, setter: setUserGitAuthProviders },
-                newSignupFlow: { defaultValue: false, setter: setNewSignupFlow },
-                linkedinConnectionForOnboarding: { defaultValue: false, setter: setLinkedinConnectionForOnboarding },
-                paymentVerificationFlow: { defaultValue: false, setter: setPaymentVerificationFlow },
-                team_only_attribution: { defaultValue: false, setter: setOrgOnlyAttribution },
-            };
-
-            for (const [flagName, config] of Object.entries(featureFlags)) {
-                const value = async () => {
-                    // First check if the flag is non-default for any of the orgs
-                    for (const org of orgs || []) {
-                        const flagValue = await getExperimentsClient().getValueAsync(flagName, config.defaultValue, {
-                            user,
-                            projectId: project?.id,
-                            teamId: org.id,
-                            teamName: org.name,
-                        });
-
-                        if (flagValue !== config.defaultValue) {
```

Comment:
Not sure what this logic does, but: What makes you confident that we can remove this "two-step logic with special case"? :thinking:

Reply:
This was [introduced for `isUsageBasedBillingEnabled`](https://github.com/gitpod-io/gitpod/commit/a4870ce5db87d35a4547afee07836e82018e50e3#diff-5f168105af652e1b7f6003d99dd45e22a5d47bf00a8665dd96abb363ac49c013) which we wanted to enable globally for the user as soon as it is enabled for one of their orgs. We fortunately do not have these kind of feature flags anymore, so we don't need the logic.

Reply:
Ah, make sense! Also checked on configcat.com, we don't seem to expect that anywhere anymore. :heavy_check_mark:

Reply:
Yep, this was a hack because we abused feature flags back then, and needed to eval independently against a list o teams